src/telemetry/mod.rs

```diff
@@ -0,0 +1,48 @@
+pub mod metrics {
```

Comment:
so why have mod inside mod like this? why not just create a top-level `metrics` mod?

Reply:
I could - my thinking was that this module is very small now and a file per module may make things too complicated. Don't mind either way though